src/openvic-simulation/pop/Pop.hpp

```diff
@@ -135,6 +136,11 @@ namespace OpenVic {
 		std::stringstream get_pop_context_text() const;
 		void reserve_needs_fulfilled_goods();
 		void fill_needs_fulfilled_goods_with_false();
+		void allocate_for_needs(
+			GoodDefinition::good_definition_map_t const& scaled_needs,
+			fixed_point_t& price_inverse_sum,
+			fixed_point_t& cash_left_to_spend
+		);
```

Comment:
```suggestion
		void allocate_for_needs(
			GoodDefinition::good_definition_map_t const& scaled_needs,
			fixed_point_t& price_inverse_sum,
			fixed_point_t& cash_left_to_spend
		) const;
```
This can be `const` as it only relies on `market_instance`, in fact it could even be a `static` function defined entirely within `Pop.cpp` if `MarketInstance const&` was passed to it as an argument (so not a `static` member function of `Pop`, but a standalone `static` function with the cpp file).

Reply:
No, `money_to_spend_per_good` is an instance variable and it needs to be mutable.